File: leetcode/0055/canJump.py
def canJump(nums: [int]) -> bool:
	n = len(nums)
	if n == 1: return True
	maxPos = 0
	for i in range(n - 1):
		if maxPos >= i:
			maxPos = max(maxPos, i + nums[i])
			if maxPos >= n-1:
				return True
	return False

print(canJump([0]))

File: leetcode/0055/canJump.swift
func canJump(_ nums: [Int]) -> Bool {
      guard nums.count > 1 else { return true }
    let n = nums.count
    var maxPos = 0
    for i in 0..<n-1 {
        if maxPos >= i {
            maxPos = max(maxPos, i+nums[i])
            if maxPos >= n-1 {
                  return true
            }
        }
    }
    return false
}
jump([2,3,1,2,4,2,3])
